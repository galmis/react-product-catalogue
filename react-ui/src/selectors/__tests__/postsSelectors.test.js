// @flow

jest.dontMock('../postsSelectors.js');

import {
  getPost,
  getTotalPosts,
  getTotalPostsPages,
  getSelectedPage,
  getPostsFromPage
} from '../postsSelectors';

const state = {
  postsById: {
    '1': 'bar',
    '2': 'foo',
    '12': 'xyz',
    '13': 'zyx'
  },
  totalRecords: 14,
  totalPages: 2,
  selectedPage: 2,
  fetchedPages: {
    '1': ['1', '2'],
    '2': ['12', '13']
  }
};

describe('postsSelectors', () => {

  it('getPost', () => {
    expect(getPost(state, '1')).toMatchSnapshot();
  });

  it('getTotalPosts', () => {
    expect(getTotalPosts(state)).toMatchSnapshot();
  });

  it('getTotalPostsPages', () => {
    expect(getTotalPostsPages(state)).toMatchSnapshot();
  });

  it('getSelectedPage', () => {
    expect(getSelectedPage(state)).toMatchSnapshot();
  });

  it('getPostsFromPage', () => {
    expect(getPostsFromPage(state, 1)).toMatchSnapshot();
  });

});
