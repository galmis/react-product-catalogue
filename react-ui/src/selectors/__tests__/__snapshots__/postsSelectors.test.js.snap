exports[`postsSelectors getPost 1`] = `"bar"`;

exports[`postsSelectors getPostsFromPage 1`] = `
Array [
  "bar",
  "foo",
]
`;

exports[`postsSelectors getSelectedPage 1`] = `2`;

exports[`postsSelectors getTotalPosts 1`] = `14`;

exports[`postsSelectors getTotalPostsPages 1`] = `2`;
